cli/command/trust/signer_remove.go

```diff
@@ -0,0 +1,137 @@
+package trust
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/docker/api/types"
+	registrytypes "github.com/docker/docker/api/types/registry"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/spf13/cobra"
+)
+
+type signerRemoveOptions struct {
+	forceYes bool
+}
+
+func newSignerRemoveCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "remove [OPTIONS] NAME IMAGE [IMAGE...]",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return removeSigner(dockerCli, args[0], args[1:], &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, signer string, images []string, options *signerRemoveOptions) error {
+	var errImages []string
+	for _, image := range images {
+		if err := removeSingleSigner(cli, image, signer, options.forceYes); err != nil {
+			fmt.Fprintln(cli.Out(), err.Error())
+			errImages = append(errImages, image)
+		}
+	}
+	if len(errImages) > 0 {
+		return fmt.Errorf("Error removing signer from: %s", strings.Join(errImages, ", "))
+	}
+	return nil
+}
+
+func isLastSignerForReleases(roleWithSig data.Role, allRoles []client.RoleWithSignatures) (bool, error) {
+	var releasesRoleWithSigs client.RoleWithSignatures
+	for _, role := range allRoles {
+		if role.Name == releasesRoleTUFName {
+			releasesRoleWithSigs = role
+			break
+		}
+	}
+	counter := len(releasesRoleWithSigs.Signatures)
+	if counter == 0 {
+		return false, fmt.Errorf("All signed tags are currently revoked, use docker trust sign to fix")
+	}
+	for _, signature := range releasesRoleWithSigs.Signatures {
+		for _, key := range roleWithSig.KeyIDs {
+			if signature.KeyID == key {
+				counter--
+			}
+		}
+	}
+	return counter < releasesRoleWithSigs.Threshold, nil
+}
+
+func removeSingleSigner(cli command.Cli, image, signerName string, forceYes bool) error {
```

Comment:
When `signerName` is part of options I think `signerName` and `forceYes` can be replaced with `options signerRemoveOptions`

Reply:
I left this function signature as is because otherwise there's a weird discrepancy between the unpacked image named and otherwise packed options.

Reply:
I don't really see how this is a problem. It's quite common for a function to only use some fields or methods on a struct.

Reply:
@dnephin: I don't think that is a safe practice: this is an inner helper function that only needs to know about one repo.  Including the extra images in the slice is extraneous info, and I worry that a future contribution may misuse it.

That said, if you feel strongly about changing to the `signerRemoveOptions`, you are the maintainer and I will make the change - just let me know.